### ISSUE TITLE
a question about train.py

### ISSUE THREAD

**Issue by foxkw** [harryhan618/SCNN_Pytorch#58]
您好，复现成功了，感谢您的工作！
我想请问一下如果想在自己的数据集上进行训练和测试，应该如何将数据集改成类似图森的格式呢，然后再传入模型训练。

**Comment by harryhan618** [comment]
我也不知道你数据集长啥样呀:joy: 。你自己了解下tusimple数据集长啥样，然后转换过去不就行了？

**Comment by foxkw** [comment]
> 我也不知道你数据集长啥样呀joy 。你自己了解下tusimple数据集长啥样，然后转换过去不就行了？

您好，我有几个疑问。1)tusimple.py的作用是把图森数据集都加载到了seg_label文件夹中吗？
2）train.py里  训练时加载的数据就来自哪里呢。  train.json？不知道在哪里定义了数据来源 `mean=(0.485, 0.456, 0.406)
std=(0.229, 0.224, 0.225)
transform_train = Compose(Resize(resize_shape), Rotation(2), ToTensor(),
                          Normalize(mean=mean, std=std)) # 数据增强
dataset_name = exp_cfg['dataset'].pop('dataset_name')  
Dataset_Type = getattr(dataset, dataset_name)
train_dataset = Dataset_Type(Dataset_Path[dataset_name], "train", transform_train)  
train_loader = DataLoader(train_dataset, batch_size=exp_cfg['dataset']['batch_size'], shuffle=True, collate_fn=train_dataset.collate, num_workers=8)# 加载`

谢谢

**Comment by harryhan618** [comment]
`dataset/Tusimple.py`是将数据集读进来。
所以你可以把你的数据处理成tusimple的格式，把标注写成tusimple类型的json，然后稍微改下`dataset/Tusimple.py`就行了。

**Comment by foxkw** [comment]
> `dataset/Tusimple.py`是将数据集读进来。
> 所以你可以把你的数据处理成tusimple的格式，把标注写成tusimple类型的json，然后稍微改下`dataset/Tusimple.py`就行了。


您好，我运行test_culane.py时报错，请问这是cpp编译出的问题吗？我觉得缺少.txt应该会自己生成并写入呀。再次打扰您很是抱歉

`loading ./experiments/exp10/exp10_best.pth ...... From Epoch:  0
  0%|                                                   | 0/542 [00:00<?, ?it/s]/home/ubuntu-502/xiong/lane/lib/python3.6/site-packages/torch/nn/parallel/_functions.py:64: UserWarning: Was asked to gather along dimension 0, but all input tensors were scalars; will instead unsqueeze and return a vector.
  warnings.warn('Was asked to gather along dimension 0, but all '
  2%|█                                         | 13/542 [00:24<10:36,  1.20s/it]Premature end of JPEG file
100%|█████████████████████████████████████████| 542/542 [09:21<00:00,  1.06it/s]
------------Configuration---------
using multi-thread, num:20
anno_dir: /media/ubuntu-502/ABC/CULane
detect_dir: /media/ubuntu-502/ABC/code/SCNN_Pytorch/experiments/exp10/coord_output/
im_dir: /media/ubuntu-502/ABC/CULane
list_im_file: /media/ubuntu-502/ABC/CULanelist/test_split/test0_normal.txt
width_lane: 30
iou_threshold: 0.5
im_width: 1640
im_height: 590
-----------------------------------
Evaluating the results...
Error: file /media/ubuntu-502/ABC/CULanelist/test_split/test0_normal.txt not exist!
------------Configuration---------
using multi-thread, num:20
anno_dir: /media/ubuntu-502/ABC/CULane
detect_dir: /media/ubuntu-502/ABC/code/SCNN_Pytorch/experiments/exp10/coord_output/
im_dir: /media/ubuntu-502/ABC/CULane
list_im_file: /media/ubuntu-502/ABC/CULanelist/test_split/test1_crowd.txt
width_lane: 30
iou_threshold: 0.5
im_width: 1640
im_height: 590
-----------------------------------
Evaluating the results...
Error: file /media/ubuntu-502/ABC/CULanelist/test_split/test1_crowd.txt not exist!
------------Configuration---------
using multi-thread, num:20
anno_dir: /media/ubuntu-502/ABC/CULane
detect_dir: /media/ubuntu-502/ABC/code/SCNN_Pytorch/experiments/exp10/coord_output/
im_dir: /media/ubuntu-502/ABC/CULane
list_im_file: /media/ubuntu-502/ABC/CULanelist/test_split/test2_hlight.txt
width_lane: 30
iou_threshold: 0.5
im_width: 1640
im_height: 590
-----------------------------------
Evaluating the results...
Error: file /media/ubuntu-502/ABC/CULanelist/test_split/test2_hlight.txt not exist!
------------Configuration---------
using multi-thread, num:20
anno_dir: /media/ubuntu-502/ABC/CULane
detect_dir: /media/ubuntu-502/ABC/code/SCNN_Pytorch/experiments/exp10/coord_output/
im_dir: /media/ubuntu-502/ABC/CULane
list_im_file: /media/ubuntu-502/ABC/CULanelist/test_split/test3_shadow.txt
width_lane: 30
iou_threshold: 0.5
im_width: 1640
im_height: 590
-----------------------------------
Evaluating the results...
Error: file /media/ubuntu-502/ABC/CULanelist/test_split/test3_shadow.txt not exist!
------------Configuration---------
using multi-thread, num:20
anno_dir: /media/ubuntu-502/ABC/CULane
detect_dir: /media/ubuntu-502/ABC/code/SCNN_Pytorch/experiments/exp10/coord_output/
im_dir: /media/ubuntu-502/ABC/CULane
list_im_file: /media/ubuntu-502/ABC/CULanelist/test_split/test4_noline.txt
width_lane: 30
iou_threshold: 0.5
im_width: 1640
im_height: 590
-----------------------------------
Evaluating the results...
Error: file /media/ubuntu-502/ABC/CULanelist/test_split/test4_noline.txt not exist!
------------Configuration---------
using multi-thread, num:20
anno_dir: /media/ubuntu-502/ABC/CULane
detect_dir: /media/ubuntu-502/ABC/code/SCNN_Pytorch/experiments/exp10/coord_output/
im_dir: /media/ubuntu-502/ABC/CULane
list_im_file: /media/ubuntu-502/ABC/CULanelist/test_split/test5_arrow.txt
width_lane: 30
iou_threshold: 0.5
im_width: 1640
im_height: 590
-----------------------------------
Evaluating the results...
Error: file /media/ubuntu-502/ABC/CULanelist/test_split/test5_arrow.txt not exist!
------------Configuration---------
using multi-thread, num:20
anno_dir: /media/ubuntu-502/ABC/CULane
detect_dir: /media/ubuntu-502/ABC/code/SCNN_Pytorch/experiments/exp10/coord_output/
im_dir: /media/ubuntu-502/ABC/CULane
list_im_file: /media/ubuntu-502/ABC/CULanelist/test_split/test6_curve.txt
width_lane: 30
iou_threshold: 0.5
im_width: 1640
im_height: 590
-----------------------------------
Evaluating the results...
Error: file /media/ubuntu-502/ABC/CULanelist/test_split/test6_curve.txt not exist!
------------Configuration---------
using multi-thread, num:20
anno_dir: /media/ubuntu-502/ABC/CULane
detect_dir: /media/ubuntu-502/ABC/code/SCNN_Pytorch/experiments/exp10/coord_output/
im_dir: /media/ubuntu-502/ABC/CULane
list_im_file: /media/ubuntu-502/ABC/CULanelist/test_split/test7_cross.txt
width_lane: 30
iou_threshold: 0.5
im_width: 1640
im_height: 590
-----------------------------------
Evaluating the results...
Error: file /media/ubuntu-502/ABC/CULanelist/test_split/test7_cross.txt not exist!
------------Configuration---------
using multi-thread, num:20
anno_dir: /media/ubuntu-502/ABC/CULane
detect_dir: /media/ubuntu-502/ABC/code/SCNN_Pytorch/experiments/exp10/coord_output/
im_dir: /media/ubuntu-502/ABC/CULane
list_im_file: /media/ubuntu-502/ABC/CULanelist/test_split/test8_night.txt
width_lane: 30
iou_threshold: 0.5
im_width: 1640
im_height: 590
-----------------------------------
Evaluating the results...
Error: file /media/ubuntu-502/ABC/CULanelist/test_split/test8_night.txt not exist!
cat: '/media/ubuntu-502/ABC/code/SCNN_Pytorch/experiments/exp10/coord_output//../evaluate/out*.txt': 没有那个文件或目录
`
